src/Core/System/Language/Command/LanguageChangeDefaultCommand.php

```diff
@@ -0,0 +1,300 @@
+            } else {
+                $locale = array_shift($localeList[$localeLang]);
+            }
+
+            $input->setArgument('locale', $locale);
+        }
+    }
+
+    /**
+     * @throws DoctrineDriverException
+     * @throws Exception
+     */
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        if ($input->isInteractive()) {
+            $helper = $this->getHelper('question');
+            $question = new ConfirmationQuestion("Are you sure you want to change the system default language?\nLoose of data is possible, please create a backup of the database! (yes/no) [yes]", true);
+            if (!$helper->ask($input, $output, $question)) {
+                return 0;
+            }
+        }
+        $localeCode = $input->getArgument('locale');
+        if (empty($localeCode)) {
+            throw new InvalidArgumentException('argument locale shouldn\'t be empty');
+        }
+        $context = new Context(new SystemSource());
+
+        $newLocale = $this->localeRepository->search(
+            (new Criteria())
+                ->addFilter(new EqualsFilter('code', $localeCode))
+                ->addAssociation('translations'),
+            $context
+        )->first();
+        if (!$newLocale instanceof LocaleEntity) {
+            $output->writeln('<error>argument locale isn\'t a valid locale code</error>');
+
+            return 1;
+        }
+
+        // if $currentDefaultLanguage is null there are more issues then we can solve
+        /** @var LanguageEntity $currentDefaultLanguage */
+        $currentDefaultLanguage = $this->languageRepository->search(
+            (new Criteria([Defaults::LANGUAGE_SYSTEM]))
+                ->addAssociation('locale'),
+            $context
+        )->first();
+
+        $currentLocaleId = $currentDefaultLanguage->getLocaleId();
+        $currentLocale = $currentDefaultLanguage->getLocale();
+        if (!$currentLocale) {
+            $output->writeln('<error>couldn\'t find current locale</error>');
+
+            return 1;
+        }
+        $newDefaultLocaleId = $newLocale->getId();
+
+        // locales match -> do nothing.
+        if ($currentLocaleId === $newDefaultLocaleId) {
+            $output->writeln(sprintf('<comment>nothing todo %s is already the system default language</comment>', $localeCode));
+
+            return 0;
+        }
+
+        $newDefaultLanguageId = $this->languageRepository->searchIds(
+            (new Criteria())
+                ->addFilter(new EqualsFilter('localeId', $newDefaultLocaleId)),
+            $context
+        )->firstId();
+
+        if (!$newDefaultLanguageId) {
+            $newDefaultLanguageId = $this->createNewLanguageEntry($newLocale, $context);
+        }
+
+        $this->swapDefaultLanguageId($newDefaultLanguageId);
+
+        $output->writeln(sprintf('<info>system default language changed to %s</info>', $newLocale->getCode()));
```

Comment:
just swapping the default language in the DB is probably not enough, as the old stuff may still be cached or saved in some denormalized structures

So we should clear the cache and rebuild all indexes after swapping the default language